We always have ffmpeg built with shared objects, so use ffmpeg as plugin
instead of builtin when possible.

Index: vlc-0.8.5-test2/configure.ac
===================================================================
--- vlc-0.8.5-test2.orig/configure.ac
+++ vlc-0.8.5-test2/configure.ac
@@ -2578,9 +2578,9 @@ dnl Trying with pkg-config
     [
      AC_CHECK_HEADERS(ffmpeg/avcodec.h)
      AC_CHECK_HEADERS(postproc/postprocess.h)
-     VLC_ADD_BUILTINS([ffmpeg])
+     VLC_ADD_PLUGINS([ffmpeg])
      if test "${enable_sout}" != "no"; then
-         VLC_ADD_BUILTINS([stream_out_switcher])
+         VLC_ADD_PLUGINS([stream_out_switcher])
      fi
      VLC_ADD_CFLAGS([ffmpeg stream_out_switcher],[${FFMPEG_CFLAGS}])
      VLC_ADD_LDFLAGS([ffmpeg],[${FFMPEG_LIBS}])
@@ -2606,10 +2606,10 @@ dnl Trying with pkg-config
         VLC_ADD_LDFLAGS([ffmpeg],[-lpostproc])])
 
       AC_CHECK_LIB(avcodec, avcodec_init, [
-        VLC_ADD_BUILTINS([ffmpeg])
+        VLC_ADD_PLUGINS([ffmpeg])
         VLC_ADD_LDFLAGS([ffmpeg],[-lavcodec])
         if test "${enable_sout}" != "no"; then
-            VLC_ADD_BUILTINS([stream_out_switcher])
+            VLC_ADD_PLUGINS([stream_out_switcher])
         fi],
          [ AC_MSG_ERROR([Could not find ffmpeg on your system: you may get it from http://ffmpeg.sf.net/ (cvs version is recommended). Alternatively you can use --disable-ffmpeg to disable the ffmpeg plugins.]) ], [$LDAVUTIL])
       AC_CHECK_LIB(avformat, av_open_input_stream, [
